### PR TITLE
Change RELEASING.md requirements

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
I'm not that convinced about `MUST NOT`, but let's start with stricter requirements and relax them if it's needed.

### CODE REVIEW

File: specification/repository.md [hunk 1]
```diff
@@ -104,8 +104,8 @@ approval is granted, GDI projects MUST NOT cut a GA release.
   - MUST have getting started information in `README.md`
   - MUST have troubleshooting information in `README.md`
   - MUST have license information in `README.md`
-- MUST have a `RELEASING.md` documenting the release process
-  - SHOULD be able to release by pushing a tag
+- MUST NOT have a `RELEASING.md` file or any release process documentation
```
Comment (pellared) [inline specification/repository.md:107]: I think that Go libraries could be an exception here.
@owais @MrAlias do you agree?

**Review by flands — COMMENTED** [review]
Overall, looks good. Should add language that all tests/CI are public and maybe a note that release a private to support signing of packages.

**Comment by mateuszrzeszutek** [comment]
> Should add language that all tests/CI are public and maybe a note that release a private to support signing of packages.

Done.

I've added a short "Exception: Go" note but I'd like to leave the actual explanation why it's okay to somebody that works with Go.

File: specification/repository.md (hunk 1 above)
Comment (owais) [reply specification/repository.md:109]: I think we should not prevent from having a RELEASING.md file. We should restrict projects from documenting internal/private release processes instead. Projects can have a RELEASING.md file and document how to do it publicly when applicable (e.g, Go libs), have it link to internal docs or simply document that release jobs are private.

**Review by owais — CHANGES_REQUESTED** [review]
I think we should not prevent from having a RELEASING.md file. We should restrict projects from documenting internal/private release processes instead. Projects can have a RELEASING.md file and document how to do it publicly when applicable (e.g, Go libs), have it link to internal docs, simply document that release jobs are private or skip adding the file altogether.

>  The release process involves signing built packages and binaries and thus must be kept private and not exposed publicly.

I think this ^ is probably enough and we don't need to ban having a RELEASING.md file specifically.

**Comment by mateuszrzeszutek** [comment]
> Projects can have a RELEASING.md file and document how to do it publicly when applicable (e.g, Go libs), have it link to internal docs, simply document that release jobs are private or skip adding the file altogether.

Good point - done.

File: specification/repository.md [hunk 2]
```diff
@@ -98,14 +98,19 @@ approval is granted, GDI projects MUST NOT cut a GA release.
...
   - SHOULD have a badge on the `README.md` with code coverage, if appropriate.
   - SHOULD have badges on the `README.md` for other relevant things including artifacts
   - MUST have getting started information in `README.md`
   - MUST have troubleshooting information in `README.md`
   - MUST have license information in `README.md`
-- MUST have a `RELEASING.md` documenting the release process
-  - SHOULD be able to release by pushing a tag
+- MAY have a `RELEASING.md` file that documents the release process, but this
+  file MUST NOT document private processes. For projects that use private release
+  jobs, the `RELEASING.md` file may be absent or just contain the following note:
```
Comment (MrAlias) [inline specification/repository.md:111]: This should be normative.

```suggestion
  jobs, the `RELEASING.md` file MAY be absent or, if included, just contain the following note:
```

**Comment by MrAlias** [comment]
@pellared @flands PTAL

**Review by owais — APPROVED** [review]

**Review by pellared — APPROVED** [review]

**Pull request merged**